www/js/logging.js

```diff
@@ -0,0 +1,119 @@
+define([
+	'dojo/_base/unload'
+], function(baseUnload){
+	var logging = {
+		_session: null,
+		instance: null,
+		setSession: function(session){
+			this._session = session;
+		},
+		//wrapper for session topomath logger
+		/**
+		* this logs each and every user events, like node opening, closing, solution entering
+		* @params:	_log: message which is a json object with all the keys and values to be logged.
+		*/
+		eventLogger: function(_log){
+			this.logger("ui-action", _log);
+		},
+		/**
+		* this is for general logging message with not a fixed type, used for logging runtime
+		* error messages and window on focus and on blur events.
+		* @params:	type - type of logging message - like "runtime-error"
+		*			_log - json object with key values that are to be logged.
+		*/
+		logger: function(type, _log){
```

Comment:
I suggest we move this function above the eventLogger function definition, for readability.